versioned_docs/version-V3/guides/swaps/single-swaps.md

```diff
@@ -4,276 +4,331 @@ title: Single Swaps
 
-    ISwapRouter public immutable swapRouter;
+```jsx
+/* Connect to WETH and wrap some eth  */
+let signers = await hre.ethers.getSigners();
+const WETH = new hre.ethers.Contract(WETH_ADDRESS, ercAbi, signers[0]);
+const deposit = await WETH.deposit({ value: hre.ethers.utils.parseEther("10") });
+await deposit.wait();
+```
 
-    // This example swaps DAI/WETH9 for single path swaps and DAI/USDC/WETH9 for multi path swaps.
+If all goes well, these steps will give us 10 WETH ready to be swapped for DAI. To begin the test, we’ll grab our test account’s current balance of DAI and save it to the local variable `DAIBalanceBefore`. You’ll notice we use the `formatUnits` utility from Ethers to convert the DAI amount to a readable floating point number: 
 
-    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
-    address public constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
-    address public constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+```jsx
+/* Check Initial DAI Balance */ 
+const DAI = new hre.ethers.Contract(DAI_ADDRESS, ercAbi, signers[0]);
+const expandedDAIBalanceBefore = await DAI.balanceOf(signers[0].address);
+const DAIBalanceBefore = Number(hre.ethers.utils.formatUnits(expandedDAIBalanceBefore, DAI_DECIMALS));
+```
 
-    // For this example, we will set the pool fee to 0.3%.
-    uint24 public constant poolFee = 3000;
+Next is a critical, often overlooked, step. Our `SimpleSwap` contract is going to be moving WETH on out of our wallet and onto the swap contract. It can’t do this without our approval, which we give by calling the `approve` method on the WETH ERC-20 contract itself. For this example, we’ll approve it to move 1 WETH on our behalf: 
 
-    constructor(ISwapRouter _swapRouter) {
-        swapRouter = _swapRouter;
-    }
+```jsx
+/* Approve the swapper contract to spend WETH for me */
+await WETH.approve(simpleSwap.address, hre.ethers.utils.parseEther("1"));
+```
 
-    /// @notice swapExactInputSingle swaps a fixed amount of DAI for a maximum possible amount of WETH9
-    /// using the DAI/WETH9 0.3% pool by calling `exactInputSingle` in the swap router.
-    /// @dev The calling address must approve this contract to spend at least `amountIn` worth of its DAI for this function to succeed.
-    /// @param amountIn The exact amount of DAI that will be swapped for WETH9.
-    /// @return amountOut The amount of WETH9 received.
-    function swapExactInputSingle(uint256 amountIn) external returns (uint256 amountOut) {
-        // msg.sender must approve this contract
+And finally we can execute our swap on the `SimpleSwap` contract. Hardhat makes this easy, exposing the callable `swapWETHForDAI` method on our contract object. We’ll use this to swap of 0.1 WETH for DAI: 
```

Comment:
Would be nice to state that it's best to calculate the gas, so as to not overspent if they make mistake in production and link to ethers 
https://docs.ethers.org/v5/api/contract/contract/#contract-estimateGas